### ISSUE TITLE
Not working in jupyter due to UnsupportedOperation error

### ISSUE THREAD

**Issue by pomo-mondreganto** [Gallopsled/pwntools#1671]
I'm working with pwntools version 4.2.1 and trying to use pwntools inside JupyterLab with python3.7 kernel. When simply importing everything with 

```
from pwn import *
```

I get the following error:

```
---------------------------------------------------------------------------
UnsupportedOperation                      Traceback (most recent call last)
<ipython-input-1-308ef5371a41> in <module>()
----> 1 from pwn import *

/usr/local/anaconda3/envs/sage/lib/python3.7/site-packages/pwn/__init__.py in <module>()
      2 from __future__ import absolute_import
      3 
----> 4 from pwn.toplevel import *
      5 
      6 pwnlib.args.initialize()

/usr/local/anaconda3/envs/sage/lib/python3.7/site-packages/pwn/toplevel.py in <module>()
     18 from pprint import pprint
     19 
---> 20 import pwnlib
     21 from pwnlib import *
     22 from pwnlib.asm import *

/usr/local/anaconda3/envs/sage/lib/python3.7/site-packages/pwnlib/__init__.py in <module>()
     41 
     42 for module in __all__:
---> 43     importlib.import_module('.%s' % module, 'pwnlib')

/usr/local/anaconda3/envs/sage/lib/python3.7/importlib/__init__.py in import_module(name, package)
    125                 break
    126             level += 1
--> 127     return _bootstrap._gcd_import(name[level:], package, level)
    128 
    129 

/usr/local/anaconda3/envs/sage/lib/python3.7/site-packages/pwnlib/args.py in <module>()
     60 import sys
     61 
---> 62 from pwnlib import term
     63 from pwnlib.context import context
     64 

/usr/local/anaconda3/envs/sage/lib/python3.7/site-packages/pwnlib/term/__init__.py in <module>()
      4 import sys
      5 
----> 6 from pwnlib.term import completer
      7 from pwnlib.term import key
      8 from pwnlib.term import keymap

/usr/local/anaconda3/envs/sage/lib/python3.7/site-packages/pwnlib/term/completer.py in <module>()
      5 import re
      6 
----> 7 from pwnlib.term import readline
      8 
      9 

/usr/local/anaconda3/envs/sage/lib/python3.7/site-packages/pwnlib/term/readline.py in <module>()
     11 from pwnlib.term import keymap as km
     12 from pwnlib.term import term
---> 13 from pwnlib.term import text
     14 
     15 cursor = text.reverse

/usr/local/anaconda3/envs/sage/lib/python3.7/site-packages/pwnlib/term/text.py in <module>()
    124 
    125 tether = sys.modules[__name__]
--> 126 sys.modules[__name__] = Module()

/usr/local/anaconda3/envs/sage/lib/python3.7/site-packages/pwnlib/term/text.py in __init__(self)
     41             'white': 7,
     42             }
---> 43         self._reset = termcap.get('reset')
     44         self._attributes = {}
     45         for x, y in [('italic'   , 'sitm'),

/usr/local/anaconda3/envs/sage/lib/python3.7/site-packages/pwnlib/term/unix_termcap.py in get(cap, *args, **kwargs)
     23 
     24     if cache is None:
---> 25         init()
     26     s = cache.get(cap)
     27     if not s:

/usr/local/anaconda3/envs/sage/lib/python3.7/site-packages/pwnlib/term/unix_termcap.py in init()
     57         # Fix for BPython
     58         try:
---> 59             curses.setupterm()
     60         except curses.error as e:
     61             import traceback

UnsupportedOperation: fileno
```

When I try to add `PWNLIB_NOTERM=true` env with 

```
%env PWNLIB_NOTERM=true
```

Using the same import I get another error:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-308ef5371a41> in <module>()
----> 1 from pwn import *

/usr/local/anaconda3/envs/sage/lib/python3.7/site-packages/pwn/__init__.py in <module>()
      2 from __future__ import absolute_import
      3 
----> 4 from pwn.toplevel import *
      5 
      6 pwnlib.args.initialize()

/usr/local/anaconda3/envs/sage/lib/python3.7/site-packages/pwn/toplevel.py in <module>()
     19 
     20 import pwnlib
---> 21 from pwnlib import *
     22 from pwnlib.asm import *
     23 from pwnlib.context import Thread

AttributeError: module 'pwnlib' has no attribute 'context'
```

I've seen some similar issues & PRs, tried using the current `dev` branch (version 4.4.0.dev0), but the error were exactly the same.

**Comment by Arusekk** [comment]
Yeah there should be some `(curses.error, UnsupportedOperation)` but this might break py2, you can try to fix that if you want.

**Comment by pomo-mondreganto** [comment]
I found out that first error (`UnsupportedOperation`) is the consequence of running the notebook inside sage math inside Jupyter, which is not related to this repo, I believe. 

The second exception, though, is more interesting as it happens in similar setup even in regular Jupyter notebooks, even though there's a merged PR #1632 that claims to fix the issue (I tested it with current `dev` branch, the problem persists).

**Comment by Arusekk** [comment]
See #1658 for some context, might probably help you

**Comment by heapcrash** [comment]
Try setting `PWNLIB_NOTERM=1` in the environment before starting Python or before importing Pwntools.

```
$ PWNLIB_NOTERM=1 python foo.py
```

```
$ python foo.py NOTERM=1
```

```py
import os
os.environ['PWNLIB_NOTERM'] = '1'
from pwn import *
```

**Comment by Arusekk** [comment]
Closed in #1659 or #1632.
